src/main.rs

```diff
@@ -1,17 +1,18 @@
-// have accesserror return more info (addr of invalid access)
-// memory w/ 3 states: unallocated, valid to write to, fully initialized
-// set stack pointer or alloc stack / free stack; designates memory as stack
+use std::cmp::*;
```

Comment:
Now that we're starting to assume invariants about memory layout, it might be good to add a doc-comment at the top here describing the assumptions. The main point now I guess is that the stack is assumed to be at the bottom of memory, right?